### PR TITLE
Feature/templatedsolverparams

### PR DESCRIPTION
**Pull request by dehann**

### CODE REVIEW

**Comment by dehann** [comment]
just hold on this briefly, getting errors in IIF now

**Comment by dehann** [comment]
fixes #29 , #30

File: src/services/GraphsDFG.jl [hunk 1]
```diff
@@ -67,7 +79,7 @@ setDescription(dfg::GraphsDFG, description::String) = dfg.description = descript
 getInnerGraph(dfg::GraphsDFG) = dfg.g
 getAddHistory(dfg::GraphsDFG) = dfg.addHistory
 getSolverParams(dfg::GraphsDFG) = dfg.solverParams
-setSolverParams(dfg::GraphsDFG, solverParams::Any) = dfg.solverParams = solverParams
+setSolverParams(dfg::GraphsDFG, solverParams) = dfg.solverParams = solverParams
```
Comment (GearsAD) [inline src/services/GraphsDFG.jl]: Can we change this to:
```
function setSolverParams(dfg::GraphsDFG, solverParams::P) where P <: AbstractParams
  dfg.solverParams = solverParams
end
```
?

File: src/services/GraphsDFG.jl [hunk 2]
```diff
@@ -125,7 +137,7 @@ function addFactor!(dfg::GraphsDFG, variables::Vector{DFGVariable}, factor::DFGF
         Graphs.add_edge!(dfg.g, edge)
     end
     # Track insertion
-    push!(dfg.addHistory, factor.label)
+    # push!(dfg.addHistory, factor.label)
```
Comment (GearsAD) [inline src/services/GraphsDFG.jl]: Is addHistory only for variables?

File: src/services/GraphsDFG.jl (hunk 1 above)
Comment (dehann) [reply src/services/GraphsDFG.jl]: done thanks

File: src/services/GraphsDFG.jl (hunk 2 above)
Comment (dehann) [reply src/services/GraphsDFG.jl]: see #32

**Pull request merged**